### ISSUE TITLE
Incorrect path separator being used

### ISSUE THREAD

**Issue by FTWinston** [jarlef/vite-plugin-preload#7]
Hi, vite-plugin-preload is writing its "preload" link entry paths using backslash as the separator, instead of forward slash.

This is occurring when building on Windows.

Without this plugin, my rendered head looks like:
```
  <head>
    <script type="module" crossorigin src="/assets/index-48601b62.js"></script>
    <link rel="stylesheet" href="/assets/index-eff2bcb1.css">
  </head>
```

With this plugin, it looks like:

```
  <head>
    <script
      type="module"
      crossorigin=""
      src="/assets/index-48601b62.js"
    ></script>
    <link rel="stylesheet" href="/assets/index-eff2bcb1.css" />
    <link rel="modulepreload" href="\assets\Deferred-b38d3a14.js" />
    <link rel="modulepreload" href="\assets\index-48601b62.js" />
    <link rel="stylesheet" href="\assets\index-eff2bcb1.css" />
  </head>
```

Because backslash is being used instead of forward slash in the "preload" entries, two problems occur:

1. The stylesheet link is written twice, with backslashes on one and forward slashes on the other.
2. The "entry point" script also has a "preload" link entry added for it, but with the wrong slashes.

In both cases, I presume it's failing to de-duplicate them successfully. I'd assume this is because path.join is using my (Windows) backslash value for [path.sep](https://nodejs.org/api/path.html#pathsep), which isn't the correct behaviour for assembling a url:

```
const path = join(base, bundle.fileName);

if (existingLinks.includes(path)) {
  continue;
}
```

**Comment by FTWinston** [comment]
A potential workaround would be to modify the call to `join` in the above snippet, like shown:

```
const path = join(base, bundle.fileName).split(path.sep).join("/");
```

**Comment by FTWinston** [comment]
With the change from the linked PR, my head now looks like this:

```
  <head>
    <script
      type="module"
      crossorigin=""
      src="/assets/index-48601b62.js"
    ></script>
    <link rel="stylesheet" href="/assets/index-eff2bcb1.css" />
    <link rel="modulepreload" href="/assets/Deferred-b38d3a14.js" />
  </head>
```

That's the exact output I'd been expecting. 🎉

**Comment by jarlef** [comment]
Thanks for the detailed issue description 👍 (and also the PR). It was a stupid oversight on my part to use the path for urls, instead of using the url module.  The fix has been deployed in version 0.3.1

**Comment by FTWinston** [comment]
No worries, it happens to us all.

Thanks for the quick turnaround!
